### PR TITLE
Update validation of imageTagName in CICD pipeline

### PR DESCRIPTION
**Pull request by nilsgstrabo**
- Do not fail deploy step in pipeline job when component/job image to deploy does not contain {imageTagName} and radixconfig has imageTagName set. This will happen when promoting.
- Fail deploy pipeline if destination environment contains components to build
- Validate components/jobs in buildeploy or deploy pipeline with {imageTagName} has imageTagName set in radixconfig or supplied as pipline argument

### CODE REVIEW

File: charts/radix-operator/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: radix-operator
-version: 1.25.4
-appVersion: 1.45.4
+version: 1.25.5
+appVersion: 1.45.5
```
Comment (Richard87) [inline charts/radix-operator/Chart.yaml:4]: Må bumpes til 7 eller over...

File: pipeline-runner/steps/errors.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+	"github.com/pkg/errors"
+)
+
+var (
+	ErrDeployOnlyPipelineDoesNotSupportBuild = errors.New("deploy pipeline does not support building components and jobs")
+)
+
+func ErrMissingRequiredImageTagName(componentName, envName string) error {
+	return fmt.Errorf("component %s requires imageTagName to be set in environmentConfig for environment %s or specified as argument to the pipeline job", componentName, envName)
```
Comment (Richard87) [inline pipeline-runner/steps/errors.go:14]: Kan deler av feilmeldingen være statisk slik at Error.Is fungerer?

**Review by Richard87 — CHANGES_REQUESTED** [review]
noen kommentarer

File: pipeline-runner/steps/errors.go (hunk 2 above)
Comment (nilsgstrabo) [reply pipeline-runner/steps/errors.go]: fixed

File: charts/radix-operator/Chart.yaml (hunk 1 above)
Comment (nilsgstrabo) [reply charts/radix-operator/Chart.yaml]: fixed

**Review by Richard87 — APPROVED** [review]
godkjent

**Pull request merged**